code/sonalyze/table/data.go

```diff
@@ -265,3 +267,288 @@ func FormatHostnames(x *Hostnames, ctx PrintMods) string {
+	}
+	return t.Unix(), nil
+}
+
+func CvtString2DateTimeValueOrBlank(s string) (any, error) {
+	return CvtString2DateTimeValue(s)
+}
+
+func CvtString2DateValue(s string) (any, error) {
+	t, err := time.Parse(time.DateOnly, s)
+	if err != nil {
+		return int64(0), err
+	}
+	return t.Unix(), nil
+}
+
+func CvtString2TimeValue(s string) (any, error) {
+	t, err := time.Parse(time.TimeOnly, s)
+	if err != nil {
+		return int64(0), err
+	}
+	return t.Unix(), nil
+}
+
+// TODO: Really needs to be case-insensitive
+func CvtString2Bool(s string) (any, error) {
+	switch s {
+	case "true", "yes", "1":
+		return true, nil
+	case "false", "no", "0":
+		return false, nil
+	default:
+		return nil, fmt.Errorf("Not a boolean value: %s", s)
+	}
+}
+
+func CvtString2Int(s string) (any, error) {
+	i, err := strconv.ParseInt(s, 10, 64)
+	if err != nil {
+		return nil, err
+	}
+	return int(i), nil
+}
+
+func CvtString2Int64(s string) (any, error) {
+	i, err := strconv.ParseInt(s, 10, 64)
+	if err != nil {
+		return nil, err
+	}
+	return i, nil
+}
+
+func CvtString2Uint8(s string) (any, error) {
+	i, err := strconv.ParseUint(s, 10, 8)
+	if err != nil {
+		return nil, err
+	}
+	return uint8(i), nil
+}
+
+func CvtString2Uint32(s string) (any, error) {
+	i, err := strconv.ParseUint(s, 10, 32)
+	if err != nil {
+		return nil, err
+	}
+	return uint32(i), nil
+}
+
+func CvtString2Uint64(s string) (any, error) {
+	i, err := strconv.ParseUint(s, 10, 64)
+	if err != nil {
+		return nil, err
+	}
+	return i, nil
+}
+
+func CvtString2Float32(s string) (any, error) {
+	i, err := strconv.ParseFloat(s, 32)
+	if err != nil {
+		return nil, err
+	}
+	return float32(i), nil
+}
+
+func CvtString2Float64(s string) (any, error) {
+	i, err := strconv.ParseFloat(s, 64)
+	if err != nil {
+		return nil, err
+	}
+	return i, nil
+}
+
+func CvtString2Hostnames(s string) (any, error) {
+	ps, err := hostglob.SplitMultiPattern(s)
+	if err != nil {
+		return nil, err
+	}
+	xs := make([]string, 0)
+	for _, p := range ps {
+		ss, err := hostglob.ExpandPattern(p)
+		if err != nil {
+			return nil, err
+		}
+		xs = append(xs, ss...)
+	}
+	return xs, nil
+}
+
+// true > false
+func CompareBool(b1, b2 bool) int {
+	if b1 == b2 {
+		return 0
+	}
+	if b1 {
+		return 1
+	}
+	return -1
+}
+
+func SetCompareGpuSets(a, b gpuset.GpuSet, op int) bool {
+	switch op {
+	case opEq:
+		return a.Equal(b)
+	case opLt:
+		return b.HasSubset(a, true)
+	case opLe:
+		return b.HasSubset(a, false)
+	case opGt:
+		return a.HasSubset(b, true)
+	case opGe:
+		return a.HasSubset(b, false)
+	default:
+		panic("Unknown op")
+	}
+}
+
+// This is special purpose. `a` comes from the data record and is assumed to have (fully) qualified
+// host names, while `b` comes from CvtString2Hostnames and will typically have just prefixes.  The
+// meaning of A<B is that every b in B is in A or is a prefix of something in A.  The meaning of A=B
+// is additionally that there are no a in A that are not matched in that way by some b in B.
```

Comment:
The following is exactly the same code as for SetCompareGpuSets, can they be merged?  The above comment is subsumed by the comments in the Hostnames abstraction and can be removed.